scipy/spatial/distance.py

```diff
@@ -1837,7 +1835,7 @@ def _cosine_cdist(XA, XB, dm):
     dm += 1
 
 
-def cdist(XA, XB, metric='euclidean', p=2, V=None, VI=None, w=None):
+def cdist(XA, XB, metric='euclidean', p=None, w=None, V=None, VI=None):
```

Comment:
I have uniformed the args order with pdist. 
Also, in the doc the arguments were presented in different order. Also, shouldn't keyword args name be always specified (at least as a good practice)?
Anyway, if you like I can revert the order.